qiling/os/memory.py

```diff
@@ -63,6 +63,43 @@ def __read_string(self, addr: int) -> str:
     def __write_string(self, addr: int, s: str, encoding: str):
         self.write(addr, bytes(s, encoding) + b'\x00')
 
+    def __getitem__(self, key: Union[slice, int]) -> bytearray:
+        if isinstance(key, slice):
+            start = key.start
+            stop = key.stop
+            step = key.step
+
+            if step is not None and step != 1:
+                # step != 1 means we have to do copy, don't allow it.
+                raise IndexError("Only support slicing continous memory")
+
+            return self.ql.mem.read(start, max(0, stop - start))
+        elif isinstance(key, int):
+            return self.ql.mem.read(key, 1)[0]
```

Comment:
This line returns an `int`, not `bytearray`, which is somewhat counterintuitive. I think you should remove the `[0]`. Also, I'm not sure why this code uses `self.ql.mem` instead of `self`. In this class, they reference / point to the same thing, right?
```suggestion
            return self.read(key, 1)
```

Reply:
> This line returns an `int`, not `bytearray`, which is somewhat counterintuitive. I think you should remove the `[0]`. 

Because we (try to) provide the same interface as `bytearray`, e.g.:

```
In [1]: a = bytearray(b"\x01\x02")

In [2]: type(a[0])
Out[2]: int

In [3]:
```

> Also, I'm not sure why this code uses `self.ql.mem` instead of `self`. In this class, they reference / point to the same thing, right?

Good point.